Modules/_functoolsmodule.c

```diff
@@ -115,25 +195,71 @@ partial_new(PyTypeObject *type, PyObject *args, PyObject *kw)
 
     pto->fn = Py_NewRef(func);
 
-    nargs = PyTuple_GetSlice(args, 1, PY_SSIZE_T_MAX);
-    if (nargs == NULL) {
+    pto->placeholder = state->placeholder;
+    if (new_nargs && PyTuple_GET_ITEM(args, new_nargs) == pto->placeholder) {
+        PyErr_SetString(PyExc_TypeError,
+                        "trailing Placeholders are not allowed");
+        return NULL;
+    }
+
+    new_args = PyTuple_GetSlice(args, 1, new_nargs + 1);
```

Comment:
Having the `new_nargs + 1` seems a bit weird but it's mainly because you did -1 first. I think it's better if don't do the +1 here.

Reply:
As per https://github.com/python/cpython/pull/119827#issuecomment-2180683652 the value represented by the variable makes sense. If that is understood, then it all is straight forward.

I added -1 to initial definition to make it clearer.